Source/Projections/Store/Converters/ProjectionsToSDKConverter.cs

```diff
@@ -29,38 +29,50 @@ public bool TryConvert<TProjection>(IEnumerable<PbCurrentState> source, out IEnu
         var currentStates = new List<CurrentState<TProjection>>();
         foreach (var protobufState in source)
         {
-            if (!TryDeserializeWithSettings<TProjection>(protobufState, out var state, out error)) return false;
+            if (!TryDeserializeWithSettings<TProjection>(protobufState, out var state, out error))
+            {
+                return false;
+            }
             currentStates.Add(state);
         }
 
         states = currentStates;
         return true;
     }
 
-    bool TryDeserializeWithSettings<TProjection>(PbCurrentState currentState, out CurrentState<TProjection> projectionState, out Exception deserializationError)
+    static bool TryDeserializeWithSettings<TProjection>(PbCurrentState currentState, out CurrentState<TProjection> projectionState, out Exception deserializationError)
         where TProjection : class, new()
     {
-        projectionState = null;
+        projectionState = default;
         var exceptionCatcher = new JsonSerializerExceptionCatcher();
         var serializerSettings = new JsonSerializerSettings { Error = exceptionCatcher.OnError };
-        var stateType = GetCurrentStateType(currentState.Type);
+        if (!TryGetCurrentStateType(currentState.Type, out var stateType, out deserializationError))
+        {
+            return false;
+        }
         var state = JsonConvert.DeserializeObject<TProjection>(currentState.State, serializerSettings);
 
         if (exceptionCatcher.Failed)
+        {
             deserializationError = new CouldNotDeserializeProjection(currentState.State, currentState.Type, currentState.Key, exceptionCatcher.Error);
-        else
-            deserializationError = null;
-
+            return false;
+        }
+        
         projectionState = new CurrentState<TProjection>(state, stateType, currentState.Key);
-
-        return !exceptionCatcher.Failed;
+        return true;
     }
 
-    CurrentStateType GetCurrentStateType(PbCurrentStateType type)
-        => type switch
+    static bool TryGetCurrentStateType(PbCurrentStateType pbType, out CurrentStateType type, out Exception error)
+    {
+        type = default;
+        error = default;
+
+        (type, error) = pbType switch
         {
-            PbCurrentStateType.CreatedFromInitialState => CurrentStateType.CreatedFromInitialState,
-            PbCurrentStateType.Persisted => CurrentStateType.Persisted,
-            _ => throw new InvalidCurrentStateType(type)
+            PbCurrentStateType.CreatedFromInitialState => (CurrentStateType.CreatedFromInitialState, default),
+            PbCurrentStateType.Persisted => (CurrentStateType.Persisted, default),
+            _ => ((CurrentStateType)pbType, new InvalidCurrentStateType(pbType))
         };
-}
\ No newline at end of file
+        return error == default;
```

Comment:
I think this would be easier to read with a switch case that sets the type or error, then returns inside the cases.